appholder/src/main/java/com/ul/ims/gmdl/appholder/fragment/ShareDocumentFragment.kt

```diff
@@ -0,0 +1,110 @@
+package com.ul.ims.gmdl.appholder.fragment
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.activity.OnBackPressedCallback
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
+import com.ul.ims.gmdl.appholder.databinding.FragmentShareDocumentBinding
+import com.ul.ims.gmdl.appholder.fragment.ShareDocumentFragmentDirections.Companion.actionShareDocumentFragmentToSelectDocumentFragment
+import com.ul.ims.gmdl.appholder.fragment.ShareDocumentFragmentDirections.Companion.actionShareDocumentFragmentToUserConsentFragment
+import com.ul.ims.gmdl.appholder.util.TransferStatus
+import com.ul.ims.gmdl.appholder.viewmodel.ShareDocumentViewModel
+
+
+class ShareDocumentFragment : Fragment() {
+    companion object {
+        private const val LOG_TAG = "ShareDocumentFragment"
+    }
+
+    private val args: ShareDocumentFragmentArgs by navArgs()
+    private lateinit var docType: String
+    private lateinit var identityCredentialName: String
+    private lateinit var userVisibleName: String
+    private var hardwareBacked = false
+
+    private var _binding: FragmentShareDocumentBinding? = null
+    private lateinit var vm: ShareDocumentViewModel
+
+    // This property is only valid between onCreateView and
+    // onDestroyView.
+    private val binding get() = _binding!!
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        docType = args.docType
+        identityCredentialName = args.identityCredentialName
+        hardwareBacked = args.hardwareBacked
+        userVisibleName = args.userVisibleName
```

Comment:
Why aren't we just storing a `Document` reference here (and also in other fragments, for example `TransferDocumentFragment`)? I mean, I imagine we'll be adding more fields to `Document` in the future which would mean extra work here..